sinks/kafka/driver.go

```diff
@@ -0,0 +1,161 @@
+	return nil
+}
+
+func (self *kafkaSink) StoreTimeseries(timeseries []sink_api.Timeseries) error {
+	if timeseries == nil || len(timeseries) <= 0 {
+		return nil
+	}
+	for _, t := range timeseries {
+		err := self.produceKafkaMessage(t, self.timeSeriesTopic)
+		if err != nil {
+			return fmt.Errorf("failed to produce Kafka messages: %s", err)
+		}
+	}
+	return nil
+}
+
+func (self *kafkaSink) StoreEvents(events []kube_api.Event) error {
+	if events == nil || len(events) <= 0 {
+		return nil
+	}
+	for _, event := range events {
+		err := self.produceKafkaMessage(event, self.eventsTopic)
+		if err != nil {
+			return fmt.Errorf("failed to produce Kafka messages: %s", err)
+		}
+	}
+	return nil
+}
+
+// produceKafkaMessage produces messages to kafka
+func (self *kafkaSink) produceKafkaMessage(v interface{}, topic string) error {
+	if v == nil {
+		return nil
+	}
+	jsonItems, err := json.Marshal(v)
+	if err != nil {
+		return fmt.Errorf("failed to transform the items to json : %s", err)
+	}
+	message := &proto.Message{Value: []byte(string(jsonItems))}
+	_, err = self.producer.Produce(topic, partition, message)
+	if err != nil {
+		return fmt.Errorf("failed to produce message to %s:%d: %s", topic, partition, err)
+	}
+	return nil
+}
+
+func (self *kafkaSink) DebugInfo() string {
+	info := fmt.Sprintf("%s\n", self.Name())
+	info += fmt.Sprintf("There are two kafka's topics: %s,%s:\n", self.eventsTopic, self.timeSeriesTopic)
+	info += fmt.Sprintf("The kafka's broker list is: %s", self.sinkBrokerHosts)
+	return info
+}
+
+func (self *kafkaSink) Name() string {
+	return "Apache-Kafka Sink"
+}
+
+func init() {
+	extpoints.SinkFactories.Register(NewKafkaSink, "kafka")
+}
+
+func NewKafkaSink(uri *url.URL, _ extpoints.HeapsterConf) ([]sink_api.ExternalSink, error) {
+	var kafkaSink kafkaSink
+	opts, err := url.ParseQuery(uri.RawQuery)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parser url's query string: %s", err)
+	}
+
+	if len(opts["timeseriestopic"]) < 1 {
+		return nil, fmt.Errorf("There is no timeseriestopic assign for config kafka-sink")
+	}
+	kafkaSink.timeSeriesTopic = opts["eventstopic"][0]
+
+	if len(opts["eventstopic"]) < 1 {
+		return nil, fmt.Errorf("There is no eventstopic assign for config kafka-sink")
+	}
+	kafkaSink.eventsTopic = opts["eventstopic"][0]
+
+	if len(opts["brokers"]) < 1 {
+		return nil, fmt.Errorf("There is no broker assign for connecting kafka broker")
+	}
+	kafkaSink.sinkBrokerHosts = append(kafkaSink.sinkBrokerHosts, opts["brokers"]...)
+
+	//connect to kafka cluster
+	brokerConf := kafka.NewBrokerConf(brokerClientID)
+	brokerConf.DialTimeout = brokerDialTimeout
+	brokerConf.DialRetryLimit = brokerDialRetryLimit
+	brokerConf.DialRetryWait = brokerDialRetryWait
+	brokerConf.LeaderRetryLimit = brokerLeaderRetryLimit
+	brokerConf.LeaderRetryWait = brokerLeaderRetryWait
+	brokerConf.AllowTopicCreation = true
+
+	broker, err := kafka.Dial(kafkaSink.sinkBrokerHosts, brokerConf)
```

Comment:
If the brokers do not respond within the timeout, will the sink setup fail? I'd expect the sink to continue setup, and keep retrying to establish connection. From the code, that is not clear to me. Can you document the behavior using comments?